src/Tools/Common/Commands/Utils.cs

```diff
@@ -59,66 +69,66 @@ public static bool ValidateArgumentsForChildProcess(int processId, string name,
-                return false;
-            }
-            else if (processId != 0 && !string.IsNullOrEmpty(port))
-            {
-                Console.WriteLine("Only one of the --process-id or --diagnostic-port options may be specified.");
-                return false;
-            }
-            else if (!string.IsNullOrEmpty(name) && !string.IsNullOrEmpty(port))
+            else if ( processId != 0 && (!string.IsNullOrEmpty(name) || !string.IsNullOrEmpty(port) || !string.IsNullOrEmpty(dsrouter))
+                    || !string.IsNullOrEmpty(name) && (!string.IsNullOrEmpty(port) || !string.IsNullOrEmpty(dsrouter))
+                    || !string.IsNullOrEmpty(port) && !string.IsNullOrEmpty(dsrouter))
             {
-                Console.WriteLine("Only one of the --name or --diagnostic-port options may be specified.");
+                Console.WriteLine("Only one of the --name, --process-id, --diagnostic-port, or --dsrouter options may be specified.");
                 return false;
             }
-            // If we got this far it means only one of --name/--diagnostic-port/--process-id was specified
+            // If we got this far it means only one of --name/--diagnostic-port/--process-id/--dsrouter was specified
             else if (!string.IsNullOrEmpty(port))
             {
                 return true;
             }
             // Resolve name option
             else if (!string.IsNullOrEmpty(name))
             {
-                processId = CommandUtils.FindProcessIdWithName(name);
-                if (processId < 0)
+                if ((processId = FindProcessIdWithName(name)) < 0)
                 {
                     return false;
                 }
             }
-            else if (processId == 0)
+            else if (!string.IsNullOrEmpty(dsrouter))
             {
-                Console.WriteLine("One of the --name, --process-id, or --diagnostic-port options must be specified when attaching to a process.");
-                return false;
+                ConsoleColor currentColor = Console.ForegroundColor;
+                Console.ForegroundColor = ConsoleColor.Yellow;
+                Console.WriteLine("WARNING: dotnet-dsrouter is a development tool not intended for production environments." + Environment.NewLine);
+                Console.ForegroundColor = currentColor;
+
+                if (dsrouter != "ios" && dsrouter != "android" && dsrouter != "ios-sim" && dsrouter != "android-emu")
+                {
+                    Console.WriteLine("Invalid value for --dsrouter. Valid values are 'ios', 'ios-sim', 'android' and 'android-emu'.");
+                    return false;
+                }
+                if ((processId = FindOrLaunchProcessWithDSRouter(dsrouter)) < 0)
+                {
+                    Console.WriteLine($"Failed to launch dsrouter: {dsrouter}. Please make sure that dotnet-dsrouter is installed and available in the PATH, and that it's not already running");
```

Comment:
Maybe we should point to how it could be installed (if we have some docs on installing global tools) and/or adding it into the error message?